wpengine/faustjs
# ⭐ feat(previews): Create `usePreviewNode` hook

## Description

Introduce the `usePreviewNode` hook for retrieving preview content for any post type. Deprecated `usePreview`.

## Related Issue(s):

- #619

## Testing

1. Checkout the `feat-usePreviewNode` PR
2. `npm install` from the monorepo root
3. `npm run build` from the monorepo root
4. Ensure current previews still function as expected (Posts, Pages)
4. Replace contents of `examples/next/getting-started/src/pages/preview.tsx` with:
    ```tsx
    import type { Page, Post } from 'client';
    import { client } from 'client';

    export default function Preview() {
      const isLoading = client.useIsLoading();
      const { typeName, node } = client.auth.usePreviewNode();

      if (isLoading || node === undefined) {
        return <p>Loading...</p>;
      }

      if (node === null) {
        return <p>Post not found</p>;
      }

      switch (typeName) {
        case 'Page': {
          const page = node as Page;
          return (
            <>
              <h1>{page.title()}</h1>
              <div dangerouslySetInnerHTML={{ __html: page.content() }} />
            </>
          );
        }
        case 'Post': {
          const post = node as Post;
          return (
            <>
              <h1>{post.title()}</h1>
              <div dangerouslySetInnerHTML={{ __html: post.content() }} />
            </>
          );
        }
        // Add custom post types here as needed
        default: {
          throw new Error(`Unknown post type: ${typeName}`);
        }
      }
    }
    ```
5. Ensure previews will new preview code function as expected (Posts, Pages, drafts, existing, etc)
5. Create a new post type in your WordPress instance
6. Create a post in that post type
7. Go to `http://localhost:3000/preview?p=xxxx&preview=true&typeName=Post` where `xxxx` is your post ID
8. View preview content

## Documentation Changes

- Deprecated `usePreview` hook notice pushing users to use `usePreviewNode` instead
